template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/atlas/AtlasKnoxRoleConfigProvider.java

```diff
@@ -38,11 +70,67 @@ public List<String> getRoleTypes() {
         return List.of("ATLAS_SERVER");
     }
 
-    @Override
-    public boolean isConfigurationNeeded(CmTemplateProcessor cmTemplateProcessor, TemplatePreparationObject source) {
+    private boolean isAtlasKnoxConfigurationNeeded(TemplatePreparationObject source) {
         return Objects.nonNull(source.getGatewayView())
                 && Objects.nonNull(source.getGatewayView().getExposedServices())
                 && source.getGatewayView().getExposedServices().contains(exposedServiceCollector.getAtlasService().getKnoxService());
     }
 
+    private boolean isSDXOptimalizationEnabled(TemplatePreparationObject source) {
+        return source.getStackType().equals(StackType.DATALAKE)
+                && isDatalakeVersionSupported(source.getBlueprintView().getBlueprintText())
+                && entitlementService.isSDXOptimizedConfigurationEnabled(ThreadBasedUserCrnProvider.getAccountId());
+    }
+
+    // Minimum version number is 7.2.8
+    private boolean isDatalakeVersionSupported(String blueprintText) {
+        if (StringUtils.isEmpty(blueprintText)) {
+            throw new IllegalArgumentException("Blueprint is empty.");
+        }
+        String versionString;
+        try {
+            versionString = objectMapper.readValue(blueprintText, CdhVersion.class).getCdhVersion();
+
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+        String[] versionNumbers = versionString.split("\\.");
+        if (versionNumbers.length < VERSION_NUMBER_LENGTH) {
+            throw new IllegalArgumentException("Version number cannot be processed.");
+        }
+        long majorVersion = Long.parseLong(versionNumbers[0]);
```

Comment:
Why not do what we do for medium duty in `SdxService`?

```
public static final String MEDIUM_DUTY_REQUIRED_VERSION = "7.2.7";
isShapeVersionSupportedByRuntime(runtime, MEDIUM_DUTY_REQUIRED_VERSION)
```

```
private boolean isShapeVersionSupportedByRuntime(String runtime, String shapeVersion) {
    // If runtime is empty, then SDX internal call was used, so we assume it's supported.
    if (StringUtils.isEmpty(runtime)) {
        return true;
    }
    Comparator<Versioned> versionComparator = new VersionComparator();
    return versionComparator.compare(() -> runtime, () -> shapeVersion) > -1;
}
```

Reply:
Thanks for the suggestion!